### PR TITLE
[KERNEL32] make some NLS and Format functions work

### PR DESCRIPTION
**Pull request by Getequ**
Wine-sync with latest version

### CODE REVIEW

**Comment by binarymaster** [comment]
Should https://github.com/reactos/reactos/blob/master/media/doc/README.WINE be updated?

```
reactos/dll/win32/kernel32/winnls/string/lang.c        # Synced in r52754
reactos/dll/win32/kernel32/winnls/string/lcformat.c    # Synced to WineStaging-3.3
reactos/dll/win32/kernel32/winnls/string/nls.c         # Synced in r52754
```

File: dll/win32/kernel32/include/kernel32.h [hunk 1]
```diff
@@ -50,6 +50,8 @@
 #define ROUND_UP(n, align) \
     ROUND_DOWN(((ULONG)n) + (align) - 1, (align))
 
+#define ARRAY_SIZE(a) (sizeof(a)/sizeof((a)[0]))
```
Comment (HBelusca) [inline dll/win32/kernel32/include/kernel32.h]: Can we move that into the concerned `/wine/` .c source files?

File: sdk/include/psdk/winnls.h [hunk 2]
```diff
@@ -978,6 +1003,12 @@ GetLocaleInfoEx(
   _Out_writes_opt_(cchData) LPWSTR lpLCData,
   _In_ int cchData);
 
+BOOL 
+WINAPI
+IsValidLocaleName(
+  LPCWSTR lpLocaleName
+);
```
Comment (HBelusca) [inline sdk/include/psdk/winnls.h]: On the same line as the last parameter.

File: dll/win32/kernel32/include/kernel32.h (hunk 1 above)
Comment (Getequ) [reply dll/win32/kernel32/include/kernel32.h]: But lang.c is laid in /winnls/ subfolder, not in /wine/

**Comment by Getequ** [comment]
@binarymaster yeah, i synched this file and in readme too

**Comment by Getequ** [comment]
Any changes required?

File: base/shell/cmd/config.h [hunk 3]
```diff
@@ -42,7 +42,7 @@
 /* Define one of these to select the used locale. */
 /*  (date and time formats etc.) used in DATE, TIME, */
 /*  DIR, PROMPT etc. */
-#define LOCALE_WINDOWS   /* System locale */
+/* #define LOCALE_WINDOWS */   /* System locale */
```
Comment (learn-more) [inline base/shell/cmd/config.h]: Can you elaborate on this change?

Comment (Getequ) [reply base/shell/cmd/config.h]: This definition never used (not found even in commented code) but his name conflicting with group of defines from winnls.h

Seems like a rudiment

Comment (learn-more) [reply base/shell/cmd/config.h]: Thanks

**Pull request merged**